tests/backends/test_control.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for the :mod:`aiida.backends.control` module."""
+import pytest
+
+from aiida.manage.manager import get_manager
+
+
+class MockRepositoryBackend():
+    """Mock of the RepositoryBackend for testing purposes."""
+
+    # pylint: disable=no-self-use
+
+    def get_info(self, *args, **kwargs):
+        """Method to return information."""
+        return 'this is information about the repo'
+
+    def delete_objects(self, *args, **kwargs):
+        """Method to delete objects."""
+
+    def maintain(self, full=False, **kwargs):
+        """Method to perform maintainance operations."""
+        text = 'this is a report on the maintenance procedure'
+        if full:
+            text += ' with the full flag on'
+        return text
+
+
+@pytest.fixture(scope='function')
+@pytest.mark.usefixtures('clear_database_before_test')
+def clear_storage_before_test():
+    """Clears the storage before a test."""
+    #-----------------------------------------------------------------#
+    # For some reason `clear_database_before_test` is not cleaning out
+    # the nodes so I have to do it manually.
```

Comment:
Have you tried just this

```suggestion
def clear_storage_before_test(clear_database_before_test):
    """Clears the storage before a test."""
```

Reply:
No I didn't try that before. Thanks, this seemed to be the problem, I was not aware that you couldn't use marks on fixtures.